diff --git a/node_modules/fast-png/lib/helpers/text.js b/node_modules/fast-png/lib/helpers/text.js
index 1c6a927..e746733 100644
--- a/node_modules/fast-png/lib/helpers/text.js
+++ b/node_modules/fast-png/lib/helpers/text.js
@@ -9,7 +9,12 @@ const crc_1 = require("./crc");
 // https://www.w3.org/TR/png/#11tEXt
 exports.textChunkName = 'tEXt';
 const NULL = 0;
-const latin1Decoder = new TextDecoder('latin1');
+let latin1Decoder;
+try {
+    latin1Decoder = new TextDecoder('latin1');
+} catch (e) {
+    latin1Decoder = new TextDecoder('utf-8');
+}
 function validateKeyword(keyword) {
     validateLatin1(keyword);
     if (keyword.length === 0 || keyword.length > 79) {
diff --git a/node_modules/fast-png/src/helpers/text.ts b/node_modules/fast-png/src/helpers/text.ts
index c752219..a12de4c 100644
--- a/node_modules/fast-png/src/helpers/text.ts
+++ b/node_modules/fast-png/src/helpers/text.ts
@@ -8,7 +8,12 @@ export const textChunkName = 'tEXt';
 
 const NULL = 0;
 
-const latin1Decoder = new TextDecoder('latin1');
+let latin1Decoder: TextDecoder;
+try {
+  latin1Decoder = new TextDecoder('latin1');
+} catch (e) {
+  latin1Decoder = new TextDecoder('utf-8');
+}
 
 function validateKeyword(keyword: string) {
   validateLatin1(keyword);
